### ISSUE TITLE
The ``simplifiable-if-expression`` message description should not only mention ``bool(test)``

### ISSUE THREAD

**Issue by LefterisJP** [PyCQA/pylint#5882]
### Bug description

Here is a simple example

```test.py
"""Simplifiable if expression false positive"""

TEST_VAR1 = True
TEST_VAR2 = 'True'
A = True if TEST_VAR1 is True else False
B = True if TEST_VAR2 == 'True' else False
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test2
test2.py:5:4: R1719: The if expression can be replaced with 'test' (simplifiable-if-expression)
test2.py:6:4: R1719: The if expression can be replaced with 'test' (simplifiable-if-expression)
```

### Expected behavior

I would only expect the first `simplifiable-if-expression` warning for `A`.

For `B` simplifying it would not work since it's comparing against the string `True` and eventually the string `False` and `bool(string)` is always `True`

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.9.10 (main, Feb  3 2022, 16:04:33) 
[GCC 11.1.0]
```

### OS / Environment

Linux 5.16.12-arch1-1 #1 SMP PREEMPT Wed, 02 Mar 2022 12:22:51 +0000 x86_64 GNU/Linux

**Comment by DanielNoord** [comment]
@LefterisJP Can't you write:

```python
B = TEST_VAR2 == 'True'
```

**Comment by LefterisJP** [comment]
> @LefterisJP Can't you write:
> 
> ```python
> B = TEST_VAR2 == 'True'
> ```

yes indeed, but what I understood for simplifiable-if-expression was telling me to use bool(): https://vald-phoenix.github.io/pylint-errors/plerr/errors/refactoring/R1719.html

Perhaps I misunderstood the error then?

**Comment by DanielNoord** [comment]
Ah, it seems like that information is incorrect. We should update the description of that message.
I'll change the issue title to reflect this. Thanks for telling us about this!

_Btw: That project (although helpful) is not the "official" `pylint` documentation. The official documentation can be found here: https://pylint.pycqa.org/en/latest/messages/refactor/simplifiable-if-expression.html.
We have been working on integrating that project and the code examples into our own documentation, see #5527_
